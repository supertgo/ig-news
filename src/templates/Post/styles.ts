import styled, { css } from 'styled-components';

export const Wrapper = styled.main`
  ${({ theme }) => css`
    max-width: 112rem;
    margin: 0 auto;
    padding: 0 ${theme.spacings.medium};
  `}
`;

export const Article = styled.article`
  max-width: 72rem;
  margin: 8rem auto 0;
`;

export const Title = styled.h1`
  font-size: 5.6rem;
  font-weight: 900;
`;

export const Time = styled.time`
  ${({ theme }) => css`
    display: block;
    font-size: ${theme.font.sizes.medium};
    color: ${theme.colors.text};
    margin-top: ${theme.spacings.small};
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.medium};
    line-height: ${theme.spacings.medium};
    font-size: 1.8rem;
    color: ${theme.colors.text};

    p,
    ul {
      margin: ${theme.spacings.small};
    }

    ul {
      padding-left: ${theme.spacings.small};

      li {
        margin: ${theme.spacings.xxsmall} 0;
      }
    }
  `}
`;
