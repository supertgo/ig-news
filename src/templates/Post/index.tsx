import Head from 'next/head';
import Base from 'templates/Base';
import * as S from './styles';

export type PostProps = {
  post: {
    slug: string;
    title: string;
    content: string;
    updatedAt: string;
  };
};

const Post = ({ post }: PostProps) => (
  <>
    <Head>
      <title>{post.title} | Ignews</title>
    </Head>
    <Base>
      <S.Wrapper>
        <S.Article>
          <S.Title>{post.title}</S.Title>
          <S.Time>{post.updatedAt}</S.Time>
          <S.Content dangerouslySetInnerHTML={{ __html: post.content }} />
        </S.Article>
      </S.Wrapper>
    </Base>
  </>
);

export default Post;
