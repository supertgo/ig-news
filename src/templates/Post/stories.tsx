import { Story, Meta } from '@storybook/react';
import Post from '.';

export default {
  title: 'Post',
  component: Post
} as Meta;

export const Default: Story = () => <Post />;
