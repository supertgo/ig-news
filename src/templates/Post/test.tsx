import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import Post from '.';

const post = {
  slug: '/posts/visoes-inovadoras-do-codigo-livre-e-do-open-source',
  title:
    'Visões inovadoras do código livre e do open source no desenvolvimento da',
  content:
    'Iniciativas de código aberto e softwares livres foram responsáveis em moldar o principal meio de comunicação moderno',
  updatedAt: '06 de fevereiro de 2022'
};

jest.mock('components/Header', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Header"></div>;
    }
  };
});

describe('<Post />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Post post={post} />);

    expect(
      screen.getByRole('heading', { name: post.title })
    ).toBeInTheDocument();

    expect(screen.getByText(post.content)).toBeInTheDocument();

    expect(screen.getByText(post.updatedAt)).toBeInTheDocument();
    expect(screen.getByTestId('Mock Header')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
