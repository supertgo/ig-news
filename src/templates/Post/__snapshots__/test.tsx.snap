// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> should render the heading 1`] = `
.c0 {
  max-width: 112rem;
  margin: 0 auto;
  padding: 0 3.2rem;
}

.c1 {
  max-width: 72rem;
  margin: 8rem auto 0;
}

.c2 {
  font-size: 5.6rem;
  font-weight: 900;
}

.c3 {
  display: block;
  font-size: 1.6rem;
  color: #A8A8B3;
  margin-top: 2.4rem;
}

.c4 {
  margin-top: 3.2rem;
  line-height: 3.2rem;
  font-size: 1.8rem;
  color: #A8A8B3;
}

.c4 p,
.c4 ul {
  margin: 2.4rem;
}

.c4 ul {
  padding-left: 2.4rem;
}

.c4 ul li {
  margin: 0.8rem 0;
}

<div
  class=""
>
  <div
    data-testid="Mock Header"
  />
  <main
    class="c0"
  >
    <article
      class="c1"
    >
      <h1
        class="c2"
      >
        Visões inovadoras do código livre e do open source no desenvolvimento da
      </h1>
      <time
        class="c3"
      >
        06 de fevereiro de 2022
      </time>
      <div
        class="c4"
      >
        Iniciativas de código aberto e softwares livres foram responsáveis em moldar o principal meio de comunicação moderno
      </div>
    </article>
  </main>
</div>
`;
